### PR TITLE
Support payment activity widget fetching & displaying differnt deposit currencies

### LINKED ISSUE #8865: Payment activity widget – fetch & display different deposit currencies
### Describe the bug
The goal of this issue is to implement whatever is needed to allow the payment widget to show different deposit currencies. Should be possible to hard-code different currencies and the widget should show the relevant data (for stores that have multiple deposit currencies).

- This PR doesn't include the UI, which is in progress elsewhere (#8765). 
- This PR includes things like…
  - Adding currency param / prop to data fetch code and react component. 
  - Sending the currency param to fetch request.
  - Some way to test it (e.g. hard-code prop in code).
- This PR will likely require server changes, so the relevant request returns data filtered by deposit currency.

### PR DESCRIPTION
**Pull request by haszari**
Fixes #8865 
Part of epic #8490

#### Changes proposed

The goal of this PR is to implement whatever is needed to allow the payment widget to show different deposit currencies. Should be possible to hard-code different currencies and the widget should show the relevant data (for stores that have multiple deposit currencies).

- This PR doesn't include the UI, which is in progress elsewhere (#8765). 
- This PR includes things like…
  - Adding currency param / prop to data fetch code and react component. 
  - Sending the currency param to fetch request.
- This PR will likely require server changes, so the relevant request returns data filtered by deposit currency. 

#### Testing instructions

Change the currencies in the code and test.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8864` or branch name `add/deposit-currency-param-for-payment-widget` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/deposit-currency-param-for-payment-widget) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 306b80a6640b33b44b001c904b400adfe1d043e8
- Build time: 2024-06-04 10:54:59 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: includes/core/server/request/class-get-reporting-payment-activity.md [hunk 1]
```diff
@@ -13,26 +13,31 @@ The `WCPay\Core\Server\Request\Get_Reporting_Payment_Activity` class is used to
...
 The `date_start` and `date_end` parameters should be in the 'YYYY-MM-DDT00:00:00' format.
 The `timezone` parameter can be passed as an offset or as a [timezone name](https://www.php.net/manual/en/timezones.php).
+`currency` should be a currency code for a store deposit currency.
 
 ## Filter
 
 When using this request, provide the following filter and arguments:
 
 - Name: `wcpay_get_payment_activity`
+- ? what
```
Comment (jessy-p) [inline includes/core/server/request/class-get-reporting-payment-activity.md]: There are no arguments for this hook. https://github.com/Automattic/woocommerce-payments/blob/develop/includes/core/server/README.md#providing-filters

File: includes/core/server/request/class-get-reporting-payment-activity.md [hunk 2]
```diff
@@ -13,26 +13,31 @@ The `WCPay\Core\Server\Request\Get_Reporting_Payment_Activity` class is used to
...
 
 ## Example:
 
 ```php
+// Does this example explain / add any value?
```
Comment (jessy-p) [inline includes/core/server/request/class-get-reporting-payment-activity.md]: It helps when someone needs a quick example. Added due to the precedent as all other classes are documented in this way.

File: includes/core/server/request/class-get-reporting-payment-activity.md (hunk 1 above)
Comment (haszari) [reply includes/core/server/request/class-get-reporting-payment-activity.md]: Reading this again I think I understand. I can see now that this documents essentially the name/endpoint of the API – aka the `hook` under the hood (inside WordPress, inside WooPayments server). 

My comments here are me musing about how we can make our API docs super clear, and focus on the most important info. The docs here explain how this PHP class is used to construct requests, which feels close to boilerplate code IMO – we'd be better off documenting the REST API.

File: includes/core/server/request/class-get-reporting-payment-activity.md (hunk 2 above)
Comment (haszari) [reply includes/core/server/request/class-get-reporting-payment-activity.md]: Do you think the precedent is worthwhile? What documentation do you think is most useful and worth maintaining? Where should those docs live?

**Comment by haszari** [comment]
Thanks for diving into my unfinished PR @jessy-p 🙌 

You're welcome to take over this PR and/or the server side PR, or restart in a clean PR – whatever works best for you. Let me know how I can assist with a clean handover – I'm keen to get myself out of the critical path for the project. 

These two PRs were me exploring how close we are to showing different deposit currency in payment widget – and I can see that we're pretty close 👌

File: includes/core/server/request/class-get-reporting-payment-activity.md (hunk 2 above)
Comment (jessy-p) [reply includes/core/server/request/class-get-reporting-payment-activity.md]: Yes, it has been useful to me several times in the past when I was trying to make a request using existing request classes. 

I think there could be improvements to the location and content of the docs. Will plan to create an RFC/ issue to discuss this.

**Review by jessy-p — APPROVED** [review]
As woocommerce-payments-server#5405 was merged, not having these changes breaks Payment Activity Card.
Reviewed and tested, working fine, so approving.

**Pull request merged**